src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/FullDefenceResponseContentProvider.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.cmc.claimstore.documents.content;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.documents.content.models.EvidenceContent;
+import uk.gov.hmcts.cmc.domain.models.PaymentDeclaration;
+import uk.gov.hmcts.cmc.domain.models.TimelineEvent;
+import uk.gov.hmcts.cmc.domain.models.evidence.DefendantEvidence;
+import uk.gov.hmcts.cmc.domain.models.response.DefenceType;
+import uk.gov.hmcts.cmc.domain.models.response.DefendantTimeline;
+import uk.gov.hmcts.cmc.domain.models.response.FullDefenceResponse;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.cmc.claimstore.utils.Formatting.formatDate;
+
+@Component
+public class FullDefenceResponseContentProvider {
+
+    public Map<String, Object> createContent(FullDefenceResponse fullDefenceResponse) {
+        requireNonNull(fullDefenceResponse);
+        Map<String, Object> content = new HashMap<>();
```

Comment:
ImmutableMap being using in `FullAdmissionResponseContentProvider` whereas `HashMap` here?
Can we make them consistent please?

Reply:
Can't use `ImmutableMap` here as values may be `null`.

Reply:
😢

Reply:
In the future we should aim to use `ImmutableMap` and not put the value if null. Not this PR though due to extra complexity and the fact that would trigger full defence retest.